specs/src/specs/resource_pricing.md

```diff
@@ -0,0 +1,248 @@
+
+
+## Gas Limit
+
+The gas limit must be included in each transaction. If the transaction exceeds
+this gas limit during the execution of the transaction, then the transaction
+will fail.
+
+> Note: When a transaction is submitted to the mempool, the transaction is not
+> fully executed. This can lead to a transaction getting accepted by the mempool
+> and eventually included in a block, yet failing because the transaction ends
+> up exceeding the gas limit.
+
+Fees are not currently refunded. While users can specify a gas price, the total
+fee is then calculated by simply multiplying the gas limit by the gas price. The
+entire fee is then deducted from the transaction no matter what.
+
+## Fee market
+
+By default, Celestia consensus nodes will use mempools that prioritize fees,
+however mempool usage cannot be enforced at the protocol level. There is
+currently no enforced minimum fee, this value is set by each consensus node in
+their `app.toml`. Transactions that do not exceed that given gas price will not
+be able to enter the that node's mempool, and thus they will also not be
+gossiped by that node.
+
+## Estimating PFB cost
+
+Generally, the gas used by a PFB transaction involves a static "fixed cost" and
+a dynamic cost based on the size of each blob involved in the transaction.
+
+> Note: For a general use case of a normal account submitting a PFB, the static
+> costs can be treated as such. However, due to the description above of how gas
+> works in the cosmos-sdk this is not always the case. Notably, if we use a
+> vesting account or the `feegrant` modules, then these static costs change.
+
+The "fixed cost" is an approximation of the gas consumed by operations outside
+the function `GasToConsume` (for example, signature verification, tx size, read
+access to accounts), which has a default value of 65,000.
+
+> Note: the first transaction sent by an account (sequence number == 0) has an
+> additional one time gas cost of 10,000. If this is the case, this should be
+> accounted for.
+
+Each blob in the PFB contributes to the total gas cost based on its size. The
+function `GasToConsume` calculates the total gas consumed by all the blobs
+involved in a PFB, where each blob's gas cost is computed by multiplying the
+blob size (in bytes) by the `gasPerByte` parameter, along with adding a static
+amount per blob.
+
+The gas cost per blob byte and gas cost per transaction byte are parameters that
+could potentially be adjusted through the system's governance mechanisms. Hence,
+actual costs may vary depending on the current settings of these parameters.
+
+## Tracing Gas Consumption
+
+This figure plots each instance of the gas meter being incremented as a colored
+dot over the execution lifecycle of a given transaction. The y-axis in units of
+gas and the x-axis is cumulative gas consumption. The legend shows which color
+indicates what the cause of the gas consumption was.
+
```

Comment:
[non-blocking]
Linking to the code used to create these diagrams would be helpful in the future if things changes or someone wants to verify.